// Copyright 2019 Zeppelin Bend Pty Ltd
// This file is part of cimbend.
// 
// cimbend is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
// 
// cimbend is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
// 
// You should have received a copy of the GNU Affero General Public License
// along with cimbend.  If not, see <https://www.gnu.org/licenses/>.


syntax = "proto3";
option java_multiple_files = true;
option java_package = "com.zepben.protobuf.cim.iec61970.base.diagramlayout";
package zepben.cim.iec61970.base.diagramlayout;

/**
 * A reference to a style used by the originating system for a diagram object.  A diagram object style describes
 * information such as line thickness, shape such as circle or rectangle etc, and color.
 */
enum DiagramObjectStyle {
    /**
     * No specific styling should be applied.
     */
    NONE = 0;
    /**
     * Diagram object should be styled as a distribution transformer.
     */
    DIST_TRANSFORMER = 1;
    /**
     * Diagram object should be styled as an isolating transformer.
     */
    ISO_TRANSFORMER = 2;
    /**
     * Diagram object should be styled as a reversible regulator transformer.
     */
    REVERSIBLE_REGULATOR = 3;
    /**
     * Diagram object should be styled as a non-reversiable transformer.
     */
    NON_REVERSIBLE_REGULATOR = 4;
    /**
     * Diagram object should be styled as a zone transformer.
     */
    ZONE_TRANSFORMER = 5;
    /**
     * Diagram object should be styled as a feeder circuit breaker.
     */
    FEEDER_CB = 6;
    /**
     * Diagram object should be styled as a circuit breaker.
     */
    CB = 7;
    /**
     * Diagram object should be styled as a junction.
     */
    JUNCTION = 8;
    /**
     * Diagram object should be styled as a disconnector.
     */
    DISCONNECTOR = 9;
    /**
     * Diagram object should be styled as a fuse.
     */
    FUSE = 10;
    /**
     * Diagram object should be styled as a recolser.
     */
    RECLOSER = 11;
    /**
     * Diagram object should be styled as a fault indicator.
     */
    FAULT_INDICATOR = 12;
    /**
     * Diagram object should be styled as a jumper.
     */
    JUMPER = 13;
    /**
     * Diagram object should be styled as a energy source.
     */
    ENERGY_SOURCE = 14;
    /**
     * Diagram object should be styled as a shunt compensator.
     */
    SHUNT_COMPENSATOR = 15;
    /**
     * Diagram object should be styled as a usage point.
     */
    USAGE_POINT = 16;
    /**
     * Diagram object should be styled as a conductor at unknown voltage.
     */
    CONDUCTOR_UNKNOWN = 17;
    /**
     * Diagram object should be styled as a conductor at low voltage.
     */
    CONDUCTOR_LV = 18;
    /**
     * Diagram object should be styled as a conductor at 6.6kV.
     */
    CONDUCTOR_6600 = 19;
    /**
     * Diagram object should be styled as a conductor at 11kV.
     */
    CONDUCTOR_11000 = 20;
    /**
     * Diagram object should be styled as a conductor at 12.7kV (SWER).
     */
    CONDUCTOR_12700 = 21;
    /**
     * Diagram object should be styled as a conductor at 22kV.
     */
    CONDUCTOR_22000 = 22;
    /**
     * Diagram object should be styled as a conductor at 33kV.
     */
    CONDUCTOR_33000 = 23;
    /**
     * Diagram object should be styled as a conductor at 66kV.
     */
    CONDUCTOR_66000 = 24;
    /**
     * Diagram object should be styled as a conductor at 132kV.
     */
    CONDUCTOR_132000 = 25;
    /**
     * Diagram object should be styled as a conductor at 220kV.
     */
    CONDUCTOR_220000 = 26;
    /**
     * Diagram object should be styled as a conductor at 275kV.
     */
    CONDUCTOR_275000 = 27;
    /**
     * Diagram object should be styled as a conductor at 500kV or above.
     */
    CONDUCTOR_500000 = 28;
}
